### ISSUE TITLE
Training Loss Not Decreasing for Audio-Visual Synchronization Module

### ISSUE THREAD

**Issue by JunZhan2000** [v-iashin/Synchformer#19]
Hi, thank you for the great work on this project!

I am trying to reproduce the training of the Audio-Visual Synchronization Module. I have trained the model on both the VGGSound dataset and our internal dataset, but the training loss consistently stays around 3 and does not decrease further.

Training Environment
torch                         2.0.0
torchaudio                    2.0.0
torchvision                   0.15.0
av                            8.1.0

Dataset: VGGSound / Internal dataset, we have resample video and audio to 25 fps and 16k sampling rate, and scaled video to 256 * x.

base batch size=32, amounts of gpu=8, base learning rate=2e-6

Questions
I have several questions regarding the training process:

1. Video Frame Rate Requirement
Is the video frame rate of 25 FPS required for both pretraining and finetuning of Syncformer? Is this a constraint imposed by Motionformer, or is there another reason? Can I use a different frame rate when re-doing the pretraining and finetuning?

2. Data Preprocessing Assumptions
Does the training code in this repository assume that the video data has already been preprocessed (i.e., video FPS, audio FPS, and video resolution are already standardized)? Is there any input validation in the training pipeline? Also, is the data normalization used during training the same as the normalization script in example.py?

3. Batch Size and Learning Rate for Finetuning
Are there specific requirements for batch size and learning rate during finetuning? Can I directly use the default settings in this repository to train on VGGSound?

Thank you in advance for your time and help! Looking forward to your response.

**Comment by JunZhan2000** [comment]
<img width="366" height="317" alt="Image" src="https://github.com/user-attachments/assets/e04b2278-f50a-4f8c-8a02-2e722e551aa4" />
This is the loss curve. After training for 8k steps, the loss barely changed. Additionally, the last checkpoint achieved a synchronization accuracy of 4.75% on VGGSound sparse clean, which is almost random. This indicates that the training completely failed to work properly, whereas the open-source script on VGGSound achieves an accuracy of 52%.

I also tried a batch size of 48×8 and gradually increased the learning rate from 2e-6 to 1e-4, but the loss still did not decrease.

Would it be possible to share the loss curve from the author's training process?

**Comment by v-iashin** [comment]
Hi, there is a bit of warming up happening before it budges. The model was fine-tuned with 32 64GB GPUs. The performance starts to improve after 4k iterations.

<img width="845" height="554" alt="Image" src="https://github.com/user-attachments/assets/9e85796a-15d7-416d-8772-90d9c38bc55b" />

1. yes, `25fps_256side_16000hz_aac`. I didn't try so I don't know.
2. yes it assumes it was processed, just like it is outlined in README.md. No, no input validation in the training pipeline, only in example.py. For normalisation, see the config: https://github.com/v-iashin/Synchformer/blob/b66668a1521d7567cc760e5544b2b5b53179b687/configs/sync.yaml#L120

I remember that it is not the batch size that impacts the training but the number of iterations.

**Comment by JunZhan2000** [comment]
> Hi, there is a bit of warming up happening before it budges. The model was fine-tuned with 32 64GB GPUs. The performance starts to improve after 4k iterations.
> 
> <img alt="Image" width="845" height="554" src="https://private-user-images.githubusercontent.com/14952597/537720455-9e85796a-15d7-416d-8772-90d9c38bc55b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg5NjczOTcsIm5iZiI6MTc2ODk2NzA5NywicGF0aCI6Ii8xNDk1MjU5Ny81Mzc3MjA0NTUtOWU4NTc5NmEtMTVkNy00MTZkLTg3NzItOTBkOWMzOGJjNTViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTIxVDAzNDQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1MmNkMjYyMWE1ZDE0NDE5MTMyNjczZDk2N2NiOWU1NDNlNmM4Y2Q4ZTdjODZlYWRjZjIzYWU0ZmFlZTJhYjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EcJFsiCMg81c7LVFiaBzWz2i5MVLqv8lOg4Q4THuRQQ">
> 1. yes, `25fps_256side_16000hz_aac`. I didn't try so I don't know.
> 2. yes it assumes it was processed, just like it is outlined in README.md. No, no input validation in the training pipeline, only in example.py. For normalisation, see the config: 
>      
>        
>          [Synchformer/configs/sync.yaml](https://github.com/v-iashin/Synchformer/blob/b66668a1521d7567cc760e5544b2b5b53179b687/configs/sync.yaml#L120)
>        
>        
>             Line 120
>          in
>          [b66668a](/v-iashin/Synchformer/commit/b66668a1521d7567cc760e5544b2b5b53179b687)
>        
>      
>      
>        
>    
>            
>              
>               transform_sequence_train:
> 
> I remember that it is not the batch size that impacts the training but the number of iterations.

Thank you. I observed a similar curve on VGGSound with the same batch size. Previously, it might have been because the data differed from the pre-training data, or the batch size was smaller so required more time.

<img width="344" height="305" alt="Image" src="https://github.com/user-attachments/assets/1aea1c09-8644-43fd-bf45-cc86a17d5fbc" />

<img width="365" height="315" alt="Image" src="https://github.com/user-attachments/assets/b03ae094-d175-41cf-9b43-52e5b88e8380" />
